### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by lexxeice**

### CODE REVIEW

File: HW_05/Aleksei_Makar/homework.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class Homework
+  attr_accessor :homework_source, :student, :pr_title, :check_status
+
+  def initialize(homework_source:, student:, pr_title:)
+    @homework_source = homework_source
+    @student = student
+    @pr_title = pr_title
+  end
+
+  def convert_to_json
```
Comment (labs-ruby) [inline HW_05/Aleksei_Makar/homework.rb]: Just `#to_json`, this is more Ruby-like syntax
```ruby
homework.to_json
```

File: HW_05/Aleksei_Makar/student.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  def initialize(name:, surname:)
+    super
+    @nickname = "#{name} #{surname}"
+  end
+
+  def create_homework(source:, title:)
+    Homework.new(homework_source: source, student: @nickname, pr_title: title)
+  end
+
+  def connect_to_api(api)
```
Comment (labs-ruby) [inline HW_05/Aleksei_Makar/student.rb]: I see you repeat `#connect_to_api` implementation in Student and Mentor classes, from my point of view you can extract this logic and put it into one place, something like Authorization class and so on

File: HW_05/Aleksei_Makar/homework.rb (hunk 1 above)
Comment (lexxeice) [reply HW_05/Aleksei_Makar/homework.rb]: correcred

File: HW_05/Aleksei_Makar/student.rb (hunk 2 above)
Comment (lexxeice) [reply HW_05/Aleksei_Makar/student.rb]: corrected

File: HW_05/Aleksei_Makar/authorization.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# autorization module
+module Authorization
+  attr_reader :connection_status, :login
+
+  def connect_to_api(api)
+    api.add_user(self)
+    @login = api
+    user_connected!(true)
```
Comment (labs-ruby) [inline HW_05/Aleksei_Makar/authorization.rb]: Why do you need `status` argument if you don't change it?

Comment (lexxeice) [reply HW_05/Aleksei_Makar/authorization.rb]: fix

**Pull request merged**